### PR TITLE
reimplemented log replaying for performance

### LINKED ISSUE #141: random generator error
[ RUN      ] core.env
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\random(2604): invalid min and max arguments for uniform_int

when testing https://github.com/imzhenyu/rDSN/pull/140

### PR DESCRIPTION
**Pull request by glglwty**
Added relevant tests in dsn.tests. The test mimics real mutation replay latency and private-log block size. New replay implementation outperforms the old one for about 2.5X.

Also fixed #141

### CODE REVIEW

File: src/apps/replication/lib/mutation_log.cpp [hunk 1]
```diff
@@ -1278,8 +1278,149 @@ int mutation_log::garbage_collection(replica_log_info_map& gc_condition)
...
+            else
+            {
+                return ERR_OK;
+            }
+        }
+    } buffers[2];
+    buffer_t* current_buffer, *next_buffer;
+
+    //number of bytes we have issued read operations
+    size_t file_dispatched_bytes;
```
Comment (imzhenyu) [inline src/apps/replication/lib/mutation_log.cpp]: use underscore as prefix for those members to differentiate with the local variables

**Comment by imzhenyu** [comment]
nice performance boosts!~

**Comment by imzhenyu** [comment]
is it possible for us to move crc checking from replay critical path to aio callback to further boost the performance?

**Comment by glglwty** [comment]
Sure. I will work on it.

**Pull request merged**